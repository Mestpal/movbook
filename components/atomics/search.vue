<template>
  <v-text-field
    label="Search"
    outlined
    single-line
    @input="saveSearch($event)"
    @keypress.enter="launchSearch"
  />
</template>

<script>
import { mapActions } from 'vuex'

export default {
  data () {
    return {
      searchInput: ''
    }
  },
  methods: {
    ...mapActions('moviesList', [
      'getMoviesListLive',
      'resetMovieListData'
    ]),
    saveSearch (event) {
      this.searchInput = event
    },
    launchSearch () {
      this.resetMovieListData()
      this.getMoviesListLive({
        query: this.searchInput
      })
      this.$router.push(`/search?q=${this.searchInput}`)
      this.searchInput = ''
    }
  }
}
</script>
