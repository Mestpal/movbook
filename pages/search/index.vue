<template>
  <v-layout
    row
    :key="reloadKey"
  >
    <v-flex>
      <div>
        <h1 v-text="'Results'" />
      </div>
      <contentsGrid
        :configuration="configuration"
        :contentsList="moviesList"
      />
    </v-flex>
  </v-layout>
</template>

<script>
import { mapGetters, mapActions } from 'vuex'
import contentsGrid from '@/components/grids/contentsGrid'

export default {
  components: {
    contentsGrid
  },
  computed: {
    ...mapGetters('moviesList', [
      'moviesList'
    ]),
    ...mapGetters('movieDBConfig', [
      'configuration'
    ])
  },
  mounted () {
    if (!this.moviesList.lenght) {
      this.getMoviesListLive({
        query: this.$route.query.q
      })
    }
  },
  methods: {
    ...mapActions('moviesList', [
      'getMoviesListLive'
    ])
  }
}
</script>
