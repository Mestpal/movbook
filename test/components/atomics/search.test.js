import Vue from 'vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import { mount, createLocalVue } from '@vue/test-utils'

import search from '@/components/atomics/search'

Vue.use(Vuetify)
Vue.use(Vuex)
const localVue = createLocalVue()

describe('search.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(search, {
      localVue
    })
  })

  it('reders the search component correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('saveSearch updates searchinput', () => {
    const event = 'test'
    wrapper.vm.saveSearch(event)
    expect(wrapper.vm.searchInput).toEqual(event)
  })

  it('launchSearch initiates search process', () => {
    wrapper.vm.resetMovieListData = jest.fn()
    wrapper.vm.getMoviesListLive = jest.fn()
    wrapper.vm.searchRedirects = jest.fn()

    wrapper.vm.launchSearch()

    expect(wrapper.vm.resetMovieListData).toHaveBeenCalled()
    expect(wrapper.vm.getMoviesListLive).toHaveBeenCalled()
    expect(wrapper.vm.searchRedirects).toHaveBeenCalled()
  })
})
