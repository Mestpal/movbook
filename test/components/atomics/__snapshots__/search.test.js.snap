// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search.vue reders the search component correctly 1`] = `
"<div class=\\"v-input theme--light v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outlined\\">
  <div class=\\"v-input__control\\">
    <div class=\\"v-input__slot\\">
      <fieldset aria-hidden=\\"true\\">
        <legend><span>â€‹</span></legend>
      </fieldset>
      <div class=\\"v-text-field__slot\\"><label for=\\"input-2\\" class=\\"v-label theme--light\\" style=\\"right: 0px; position: absolute;\\">Search</label><input id=\\"input-2\\" type=\\"text\\"></div>
    </div>
    <div class=\\"v-text-field__details\\">
      <div class=\\"v-messages theme--light\\">
        <div class=\\"v-messages__wrapper\\"></div>
      </div>
    </div>
  </div>
</div>"
`;
